### PR TITLE
Adding Protobufs for CorfuStore [PART 1]

### PR DESCRIPTION
**Pull request by zalokhan**
## Overview

Description:
Introducing the relevant protobufs required to initialize the CorfuStore.

Why should this be merged: 
Introducing CorfuStore.

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

File: runtime/proto/corfu_options.proto [hunk 1]
```diff
@@ -0,0 +1,18 @@
+syntax = "proto2";
```
Comment (pankti-m) [inline runtime/proto/corfu_options.proto]: why proto2 and not 3?

File: test/src/test/resources/proto/test_schema.proto [hunk 2]
```diff
@@ -0,0 +1,24 @@
+syntax = "proto2";
+
```
Comment (xiaoqin2012) [inline test/src/test/resources/proto/test_schema.proto]: Same here, why someplace use proto3 some place uses proto2?

**Comment by zalokhan** [comment]
@pankti-m @xiaoqin2012 Addressed your comments.
Changed all Corfu options and metadata to use proto3.
The test protobuf classes will continue to use proto2 to test the backward compatibility.

File: runtime/proto/corfu_options.proto [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+// Option tags to be used by the CorfuStore consumers to tag special fields to be detected by Corfu.
+message SchemaOptions {
+    // Secondary keys to be indexed.
+    bool secondary_key = 1;
+    // Version number in metadata field.
+    bool version = 2;
+}
+
+extend google.protobuf.FieldOptions {
+    SchemaOptions schema = 1036;
```
Comment (lixinchengdu) [inline runtime/proto/corfu_options.proto]: Can someone explain to me what the magic number 1036 means here?

File: test/src/test/resources/proto/test_schema.proto [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+package org.corfudb.test;
+option java_package = "org.corfudb.test";
+
+import "corfu_options.proto";
+
+message Uuid {
+    optional uint64 msb = 1;
```
Comment (lixinchengdu) [inline test/src/test/resources/proto/test_schema.proto]: Can I know why the both msb and lsb are optional? Does it means that if one field is missing, the value of that filed is 0?

File: runtime/proto/corfu_options.proto (hunk 3 above)
Comment (hisundar) [reply runtime/proto/corfu_options.proto]: It is google protobuf's FieldOptions field number that does not collide with an existing number.

File: test/src/test/resources/proto/test_schema.proto (hunk 4 above)
Comment (hisundar) [reply test/src/test/resources/proto/test_schema.proto]: "optional" is the default specifier in proto3 
In this case it should really be "required" but this test is highlighting the possibility of defining protobufs in this manner.

File: runtime/proto/corfu_options.proto (hunk 3 above)
Comment (medhavidhawan) [inline runtime/proto/corfu_options.proto]: please add a comment

File: runtime/proto/corfu_store_metadata.proto [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+message TableDescriptors {
+    google.protobuf.DescriptorProto keyDescriptor = 1;
+    google.protobuf.DescriptorProto valueDescriptor = 2;
+}
+
+// Metadata.
+message TableMetadata {
+    TableName table_name = 1;
+    bool disk_based = 2;
+    bool cache = 3;
```
Comment (medhavidhawan) [inline runtime/proto/corfu_store_metadata.proto]: please add comment for this

File: runtime/proto/corfu_options.proto (hunk 3 above)
Comment (zalokhan) [reply runtime/proto/corfu_options.proto]: Done.

File: runtime/proto/corfu_store_metadata.proto (hunk 5 above)
Comment (zalokhan) [reply runtime/proto/corfu_store_metadata.proto]: Done.

**Pull request merged**